src/WebUI/WorkLoads/WorkloadDetails.tsx

```diff
@@ -0,0 +1,207 @@
+*/
+
+import { BaseComponent } from "@uifabric/utilities";
+import { localeFormat, format } from "azure-devops-ui/Core/Util/String";
+import { LabelGroup, WrappingBehavior } from "azure-devops-ui/Label";
+import { ITableColumn, SimpleTableCell as renderTableCell, Table } from "azure-devops-ui/Table";
+import { ObservableValue } from "azure-devops-ui/Core/Observable";
+import * as Date_Utils from "azure-devops-ui/Utilities/Date";
+import { ArrayItemProvider } from "azure-devops-ui/Utilities/Provider";
+import * as React from "react";
+import * as Resources from "../Resources";
+import { IVssComponentProperties } from "../Types";
+import { Utils } from "../Utils";
+import "../Services/ServiceDetails.scss";
+import { PodsTable } from "../Pods/PodsTable";
+import { PodsDetails } from "../Pods/PodsDetails";
+import { ResourceStatus } from "../Common/ResourceStatus";
+import { StoreManager } from "../FluxCommon/StoreManager";
+import { BaseKubeTable } from "../Common/BaseKubeTable";
+import { IStatusProps, Statuses, StatusSize } from "azure-devops-ui/Status";
+import { V1Pod, V1ObjectMeta, V1PodTemplateSpec } from "@kubernetes/client-node";
+import "./WorkloadDetails.scss";
+import { PodsStore } from "../Pods/PodsStore";
+import { KubeZeroData } from "../Common/KubeZeroData";
+
+export interface IWorkloadDetailsProperties extends IVssComponentProperties {
+    parentMetaData: V1ObjectMeta;
+    podTemplate: V1PodTemplateSpec;
+    parentKind: string;
+    statusProps?: IStatusProps;
+}
+
+export interface IWorkloadDetailsState {
+    pods: Array<V1Pod>;
+    selectedPod: V1Pod | null;
+    showSelectedPod: boolean;
+}
+
+export class WorkloadDetails extends BaseComponent<IWorkloadDetailsProperties, IWorkloadDetailsState> {
+    constructor(props: IWorkloadDetailsProperties) {
+        super(props, {});
+        this.state = {
+            pods: [],
+            selectedPod: null,
+            showSelectedPod: false
+        };
+        this._podsStore = StoreManager.GetStore<PodsStore>(PodsStore);
+    }
+
+    public render(): JSX.Element {
+        if (this.state.selectedPod && this.state.showSelectedPod) {
+            const parentName = this.props.parentMetaData.name || "";
+            return (<PodsDetails
+                pods={this.state.pods}
+                parentName={parentName}
+            />);
+        }
+
+        return (
+            <div className="workload-details-content">
+                {this._getMainHeading()}
+                {this._getWorkloadDetails()}
+                {this._getAssociatedPods()}
+            </div>
+        );
+    }
+
+    public componentDidMount(): void {
+        const podList = this._podsStore.getState().podsList;
+        let pods: V1Pod[] = (podList && podList.items || []).filter(pod => {
+            return Utils.isOwnerMatched(pod.metadata, this.props.parentMetaData.uid);
+        });
+
+        this.setState({
+            pods: pods,
+            selectedPod: pods[0]
```

Comment:
make sure this [0] is not an issue, check for length

Reply:
done